pkg/controller/utils/secrets_test.go

```diff
@@ -0,0 +1,29 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TesCheckSums(t *testing.T) {
+	tests := []struct {
+		name     string
+		jsonStr1 string
```

Comment:
Add a subtest that verifies that the checksums are different given different inputs.